mediapipe/tasks/python/vision/gesture_recognizer.py

```diff
@@ -0,0 +1,467 @@
+_NORM_RECT_TAG = 'NORM_RECT'
+_HAND_GESTURE_STREAM_NAME = 'hand_gestures'
+_HAND_GESTURE_TAG = 'HAND_GESTURES'
+_HANDEDNESS_STREAM_NAME = 'handedness'
+_HANDEDNESS_TAG = 'HANDEDNESS'
+_HAND_LANDMARKS_STREAM_NAME = 'landmarks'
+_HAND_LANDMARKS_TAG = 'LANDMARKS'
+_HAND_WORLD_LANDMARKS_STREAM_NAME = 'world_landmarks'
+_HAND_WORLD_LANDMARKS_TAG = 'WORLD_LANDMARKS'
+_TASK_GRAPH_NAME = 'mediapipe.tasks.vision.gesture_recognizer.GestureRecognizerGraph'
+_MICRO_SECONDS_PER_MILLISECOND = 1000
+
+
+@dataclasses.dataclass
+class GestureRecognitionResult:
+  """The gesture recognition result from GestureRecognizer, where each vector
+  element represents a single hand detected in the image.
+
+  Attributes:
+    gestures:  Recognized hand gestures with sorted order such that the
+      winning label is the first item in the list.
+    handedness: Classification of handedness.
+    hand_landmarks: Detected hand landmarks in normalized image coordinates.
+    hand_world_landmarks: Detected hand landmarks in world coordinates.
+  """
+
+  gestures: List[List[category_module.Category]]
+  handedness: List[List[category_module.Category]]
+  hand_landmarks: List[List[landmark_module.NormalizedLandmark]]
+  hand_world_landmarks: List[List[landmark_module.Landmark]]
+
+
+@dataclasses.dataclass
+class GestureRecognizerOptions:
+  """Options for the gesture recognizer task.
+
+  Attributes:
+    base_options: Base options for the hand gesture recognizer task.
+    running_mode: The running mode of the task. Default to the image mode.
+      Gesture recognizer task has three running modes:
+      1) The image mode for recognizing hand gestures on single image inputs.
+      2) The video mode for recognizing hand gestures on the decoded frames of a
+         video.
+      3) The live stream mode for recognizing hand gestures on a live stream of
+         input data, such as from camera.
+    num_hands: The maximum number of hands can be detected by the recognizer.
+    min_hand_detection_confidence: The minimum confidence score for the hand
+      detection to be considered successful.
+    min_hand_presence_confidence: The minimum confidence score of hand presence
+      score in the hand landmark detection.
+    min_tracking_confidence: The minimum confidence score for the hand tracking
+      to be considered successful.
+    min_gesture_confidence: The minimum confidence score for the gestures to be
+      considered successful. If < 0, the gesture confidence thresholds in the
+      model metadata are used.
+      TODO: Note this option is subject to change, after scoring merging
+      calculator is implemented.
+    result_callback: The user-defined result callback for processing live stream
+      data. The result callback should only be specified when the running mode
+      is set to the live stream mode.
+  """
+  base_options: _BaseOptions
+  running_mode: _RunningMode = _RunningMode.IMAGE
+  num_hands: Optional[int] = 1
```

Comment:
As those parameters having default values, do we still need them to be Optional?

Reply:
@jiuqiant Yes, for the gesture recognizer API to work properly they need to be made optional.

Reply:
I mean the category options here
mediapipe/tasks/python/components/containers/category.py

Reply:
As for the options here, I believe they don't need any default values.

Reply:
I've aligned it with the Java API FYI

https://github.com/google/mediapipe/blob/313b387cedb5b3357139ff0c14f1f847bece9512/mediapipe/tasks/java/com/google/mediapipe/tasks/vision/gesturerecognizer/GestureRecognizer.java#L449